lib/src/configuration.dart

```diff
@@ -429,18 +429,22 @@ extension SchemaObjectInternal on SchemaObject {
 /// This can happen if the server is rolled back or restored from backup.
 /// {@category Sync}
 abstract class ClientResetHandler {
-  /// Defines what should happen in case of a Client reset
-  ClientResyncModeInternal get _mode;
+  
+  // Defines what should happen in case of a Client reset
+  final ClientResyncModeInternal _mode;
 
-  BeforeResetCallback? get _onBeforeReset => null;
-  AfterResetCallback? get _onAfterDiscard => null;
-  AfterResetCallback? get _onAfterRecovery => null;
+  final BeforeResetCallback? _onBeforeReset;
+  final AfterResetCallback? _onAfterDiscard;
+  final AfterResetCallback? _onAfterRecovery;
 
   /// The callback that handles the [ClientResetError].
   final ClientResetCallback? onManualReset;
 
   /// Initializes a new instance of [ClientResetHandler].
-  const ClientResetHandler(this.onManualReset);
+  const ClientResetHandler._(this.onManualReset,  this._mode, {BeforeResetCallback? onBeforeReset, AfterResetCallback? onAfterDiscard, AfterResetCallback? onAfterRecovery})
```

Comment:
```suggestion
const ClientResetHandler._(this._mode, this.onManualReset, {BeforeResetCallback? onBeforeReset, AfterResetCallback? onAfterDiscard, AfterResetCallback? onAfterRecovery})
```
I prefer to start with the mode and them to provide all the callBacks.

Reply:
I'm OK with the rest of the other changes in the Configuration. The API doc also looks as good as before, so that we keep the user API the same.

Reply:
fixed